src/reducers/versions.tsx

```diff
@@ -341,13 +368,18 @@ export type VersionsState = {
       [path: string]: boolean;
     };
   };
+  versionEntriesIsLoading: {
+    [versionEntriesKey: string]: boolean;
+  };
 };
 
 export const initialState: VersionsState = {
   byAddonId: {},
   compareInfo: {},
   compareInfoIsLoading: {},
   currentVersionId: undefined,
+  versionEntries: {},
```

Comment:
Ah, yes, I see what you mean that the patch has grown too large. We try to keep patches around 400-600 lines of changes to make reviewing easier but it's not a hard limit.

In this case, I suggest trying to land the first patch just with the introduction of the new `versionEntries` state key. This will require a lot of churn in other components so let's focus on that first. In other words, try to remove `beginFetchVersionEntries`, `abortFetchVersionEntries`, and anything else unrelated. And, yeah, it would be ok to throw an error when entries have not been loaded just for this first patch.

In the next patch, you can re-introduce the new actions you need as well as the thunk for fetching.

Do you think a smaller patch as I described will work?

Reply:
Thanks for the suggestion. I removed those changes (still 1000+ lines changes....)

Reply:
This looks a lot better for reviewing, thanks. I'll take a closer look.